app/policies/assignment_policy.rb

```diff
@@ -33,6 +33,12 @@ def reopen?
     true
   end
 
+  def close?
+    raise CustomAuthError, "Project is already closed" if assignment.status == "closed"
+
+    true
```

Comment:
This allows all users to close assignment, which is a big security hole

Reply:
May you please help me out, how to solve this !
as far as I can see,

other members have no option to close assignment  
<img width="537" alt="Screenshot 2022-01-11 at 5 25 20 PM" src="https://user-images.githubusercontent.com/76155456/148939719-0ebf1adb-6430-4691-bce1-43b17e44236d.png">

Thanks in advance 🤗

Reply:
You are just hiding it in the ui, an attacker can goto the link even if it is not enabled on the ui